partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -546,45 +555,10 @@ internal object RexConverter {
-                is Type.NullType -> rex(StaticType.NULL, call("cast_null", arg0))
-                is Type.Missing -> rex(StaticType.MISSING, call("cast_missing", arg0))
-                is Type.Bool -> rex(StaticType.BOOL, call("cast_bool", arg0))
-                is Type.Tinyint -> TODO("Static Type does not have TINYINT type")
-                is Type.Smallint, is Type.Int2 -> rex(StaticType.INT2, call("cast_int16", arg0))
-                is Type.Int4 -> rex(StaticType.INT4, call("cast_int32", arg0))
-                is Type.Bigint, is Type.Int8 -> rex(StaticType.INT8, call("cast_int64", arg0))
-                is Type.Int -> rex(StaticType.INT, call("cast_int", arg0))
-                is Type.Real -> TODO("Static Type does not have REAL type")
-                is Type.Float32 -> TODO("Static Type does not have FLOAT32 type")
-                is Type.Float64 -> rex(StaticType.FLOAT, call("cast_float64", arg0))
-                is Type.Decimal -> rex(StaticType.DECIMAL, call("cast_decimal", arg0))
-                is Type.Numeric -> rex(StaticType.DECIMAL, call("cast_numeric", arg0))
-                is Type.Char -> rex(StaticType.CHAR, call("cast_char", arg0))
-                is Type.Varchar -> rex(StaticType.STRING, call("cast_varchar", arg0))
-                is Type.String -> rex(StaticType.STRING, call("cast_string", arg0))
-                is Type.Symbol -> rex(StaticType.SYMBOL, call("cast_symbol", arg0))
-                is Type.Bit -> TODO("Static Type does not have Bit type")
-                is Type.BitVarying -> TODO("Static Type does not have BitVarying type")
-                is Type.ByteString -> TODO("Static Type does not have ByteString type")
-                is Type.Blob -> rex(StaticType.BLOB, call("cast_blob", arg0))
-                is Type.Clob -> rex(StaticType.CLOB, call("cast_clob", arg0))
-                is Type.Date -> rex(StaticType.DATE, call("cast_date", arg0))
-                is Type.Time -> rex(StaticType.TIME, call("cast_time", arg0))
-                is Type.TimeWithTz -> rex(TimeType(null, true), call("cast_timeWithTz", arg0))
-                is Type.Timestamp -> TODO("Need to rebase main")
-                is Type.TimestampWithTz -> rex(StaticType.TIMESTAMP, call("cast_timeWithTz", arg0))
-                is Type.Interval -> TODO("Static Type does not have Interval type")
-                is Type.Bag -> rex(StaticType.BAG, call("cast_bag", arg0))
-                is Type.List -> rex(StaticType.LIST, call("cast_list", arg0))
-                is Type.Sexp -> rex(StaticType.SEXP, call("cast_sexp", arg0))
-                is Type.Tuple -> rex(StaticType.STRUCT, call("cast_tuple", arg0))
-                is Type.Struct -> rex(StaticType.STRUCT, call("cast_struct", arg0))
-                is Type.Any -> rex(StaticType.ANY, call("cast_any", arg0))
-                is Type.Custom -> TODO("Custom type not supported ")
-            }
+            val argType = arg0.type.toRuntimeType()
```

Comment:
We have to use a resolved/unresolved cast variant. We don't know the type of the arg yet so we can't insert a cast. See https://github.com/partiql/partiql-lang-kotlin/blob/partiql-spi-planner/partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt#L590